tests/ui/stability-attribute/stability-attribute-sanity.stderr

```diff
@@ -91,26 +91,29 @@ LL | #[unstable(feature = "b", issue = "none")]
 error[E0544]: multiple stability levels
   --> $DIR/stability-attribute-sanity.rs:57:1
    |
-LL | #[stable(feature = "a", since = "b")]
+LL | #[stable(feature = "a", since = "4.4.4")]
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: 'since' must be a Rust version number, such as "1.31.0"
+  --> $DIR/stability-attribute-sanity.rs:60:1
+   |
+LL | #[stable(feature = "e", since = "b")]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 error[E0544]: multiple stability levels
-  --> $DIR/stability-attribute-sanity.rs:64:1
+  --> $DIR/stability-attribute-sanity.rs:65:1
    |
 LL | #[rustc_const_unstable(feature = "d", issue = "none")]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-error: invalid stability version found
```

Comment:
This older error comes from:

https://github.com/rust-lang/rust/blob/4af886f8ab94543caad689ee6bf6a93fa8bd4a98/compiler/rustc_passes/src/stability.rs#L235

It only kicked in if there was also a `deprecated` attribute on the same item.

I might follow up with a separate PR to store a more structured representation for the `since` value than what's currently present in `StabilityLevel::Stable`. We can eliminate the reparsing done by rustc_passes and make that error unreachable.

https://github.com/rust-lang/rust/blob/4af886f8ab94543caad689ee6bf6a93fa8bd4a98/compiler/rustc_attr/src/builtin.rs#L185-L187

It would also be good if `since` could store an `ExprKind::Err`-like variant, because currently if `since` is invalid, rustc goes forward pretending the whole `stable` attribute does not exist, giving the bogus extra error _"function has missing stability attribute"_.

Reply:
#117148